profiler/src/main/java/com/splunk/opentelemetry/profiler/EventProcessingChain.java

```diff
@@ -39,15 +44,36 @@ void accept(RecordedEvent event) {
     String eventName = event.getEventType().getName();
     switch (eventName) {
       case ContextAttached.EVENT_NAME:
-        spanContextualizer.updateContext(event);
-        break;
       case ThreadDumpProcessor.EVENT_NAME:
-        threadDumpProcessor.accept(event);
+        buffer.add(event);
         break;
       case TLABProcessor.NEW_TLAB_EVENT_NAME:
       case TLABProcessor.OUTSIDE_TLAB_EVENT_NAME:
         tlabProcessor.accept(event);
         break;
     }
   }
+
+  /**
+   * Tells the processing chain that a work unit (JFR file) is complete and it can process what's in
+   * the buffer. After flushing, the buffer will be empty.
+   */
+  public void flushBuffer() {
+    buffer.forEach(dispatchContextualizedThreadDumps());
```

Comment:
interesting that we do these one at a time. Could any efficiencies be gained by dumping the whole buffer at once?

Reply:
I'm not sure I follow. Interesting that we do _what_ one at a time? File handling? Event handling? 

The buffer has to be replayed sequentially in order to know what threads where handling which spans at the time of a thread dump (the `SpanContextualizer`)....I don't think that can be done in bulk or parallelized.

Reply:
There's a `removeAll` on the PriorityQueue. If you do that, I wonder if the queue becomes available more quickly, than if you pull the items one at a time. That's all I meant.

Reply:
Ah, ok. Seems like a micro optimization if anything.

Reply:
yeah... it's just a thought. Might not apply to anything here, but there might be optimizations internally that we're missing out on if we pull them externally one at a time. Should be benchmarked before being changed, obviously.